### PR TITLE
Update api docs to handle job put config

### PR DESCRIPTION
**Pull request by arnthor-agustsson**
* Add Job with config response
* Add route for job put
* Update get job to show correct result

# Jira Ticket

[https://spotinst.atlassian.net/browse/BGD-2462](https://spotinst.atlassian.net/browse/BGD-2462)

# Demo

_Please add a recording of the feature/bug fix in work. if you added new routes, the recording should show the request and response for each new/changed route_

# Checklist:
- [ ] I have filled relevant self assessment ([NodeJS](https://docs.google.com/forms/d/e/1FAIpQLSfl14u9AOBAmxVJ272tvO7XNuXE-EMvWaGcaRZalu1UAKB7RA/viewform), [Frontend](https://docs.google.com/forms/d/e/1FAIpQLSdiBPNKH81w_EkavihVL8Uwb0j7tP8PwJmLFYm2nCOQxz-1qw/viewform), [Backend](https://docs.google.com/forms/d/e/1FAIpQLSed_PsTJ5-XIWkFL6BSDE2AQRBVPmwc3PAmHZkUn-erzVI37Q/viewform?usp=sf_link))
- [x] I have run ESlint on my changes and fixed all warnings and errors (**NodeJS & Frontend Services**)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have validated all the requirements in the Jira task were answered
- [x] I have all neccessary approvals for the design/mini design of this task
- [x] I have approved the API changes and granular permission patterns (documentation subtask) (**For public services only**)

### CODE REVIEW

File: api/services/ocean/spark/schemas/oceanSparkJobWithConfig.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  lastApp:
+    $ref: "oceanSparkApplication.yaml"
+  history:
+    description: "The most recent Spark applications in this job."
+    type: array
+    items:
+      $ref: "oceanSparkApplication.yaml"
+  config:
+    $ref: "oceanSparkConfigTemplate.yaml"
+  trends:
```
Comment (epignot) [inline api/services/ocean/spark/schemas/oceanSparkJobWithConfig.yaml:39]: I think `trends` is not exposed in the endpoint

File: api/services/ocean/spark/schemas/oceanSparkJobWithConfig.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    $ref: "oceanSparkClusterId.yaml"
+  createdAt:
+    # I had to write createdAt and updatedAt manually and not use createdAtUpdatedAt.yaml.
+    # Otherwise the object is marked as "recursive". This looks like a redoc bug.
+    $ref: "createdAt.yaml"
+  updatedAt:
+    $ref: "updatedAt.yaml"
+  lastApp:
+    $ref: "oceanSparkApplication.yaml"
+  history:
```
Comment (epignot) [inline api/services/ocean/spark/schemas/oceanSparkJobWithConfig.yaml:32]: I think `lastApp` and `history` are not populated by the db query used

File: api/services/ocean/spark/paths/jobConfig.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    - $ref: "../parameters/path/oceanSparkClusterId.yaml"
+    - $ref: "../parameters/path/oceanSparkJobId.yaml"
+  requestBody:
+    required: false # if empty then the config is deleted 
+    content:
+      application/json:
+        schema:
+          type: object
+          properties:
+            displayName:
```
Comment (epignot) [inline api/services/ocean/spark/paths/jobConfig.yaml:19]: Looking at the model used by the router, I don't think you can pass a `displayName` in the request body

**Review by epignot — COMMENTED** [review]
I think there is a mismatch for the config type in the response, and you could drop some of the response fields which will never be returned.
Other than that, it seems to me your changes fit with the existing docs 👍

File: api/services/ocean/spark/schemas/oceanSparkJobWithConfig.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  clusterId:
+    $ref: "oceanSparkClusterId.yaml"
+  createdAt:
+    # I had to write createdAt and updatedAt manually and not use createdAtUpdatedAt.yaml.
+    # Otherwise the object is marked as "recursive". This looks like a redoc bug.
+    $ref: "createdAt.yaml"
+  updatedAt:
+    $ref: "updatedAt.yaml"
+  config:
+    $ref: "ceanSparkApplicationSpec.yaml"
```
Comment (epignot) [inline api/services/ocean/spark/schemas/oceanSparkJobWithConfig.yaml:31]: small typo
```suggestion
    $ref: "oceanSparkApplicationSpec.yaml"
```

**Review by epignot — COMMENTED** [review]
Small typo, 
otherwise LGTM

**Review by epignot — APPROVED** [review]
LGTM ✅

**Review by jrj-d — APPROVED** [review]

**Pull request merged**